docs/cli/ilab-rag-retrieval.md

```diff
@@ -0,0 +1,264 @@
+# Ingesting and Utilizing RAG Artifacts Design Proposal
+
+**Author**: Daniele Martinoli
+
+**Version**: 0.1
+
+## 1. Introduction
+Purpose of this document is to propose `ilab` CLI changes to extend the InstructLab flow to generate and utilize RAG artifacts
+in the customer workflows. The proposal includes new commands to ingest a local RAG from pre-processed customer documentation, that we assume to be generated from docling output, and updates to the existing commands to utilize these RAG artifacts.
+
+## 2. Use Case
+* Accelerate POCs of fine-tuning LLMs + RAG.
+* Generate artifacts that data scientists can use to build custom solutions in their development environment:
+  * Jupyter Notebooks & Python Scripts
+  * Podman Desktop
+  * Red Hat OpenShift AI
+  * Internal Red Hat CI systems for products or services (e.g., Lightspeed products)
+
+## 3. Proposed Commands
+**Note**: In the context of version 1.4, currently under development, no changes to the command-line interface should be
+allowed. Therefore, we also propose alternative approaches to run the same RAG pipelines using existing `ilab` commands or
+other provided tools.
+
+### 3.1 RAG Ingestion Pipeline Command
+The proposal is to add a `rag` subgroup under the `data` group, with an `ingest` command, like:
+```
+ilab data rag ingest /path/to/docs/folder
+```
+
+#### Command Purpose
+This command processes embeddings generated from documents located in the */path/to/docs/folder* folder and stores them in a vector database. These embeddings are intended to be used as augmented context in a Retrieval-Augmented Generation (RAG) chat pipeline.
+
+#### Assumptions
+The documents must be in JSON format and pre-processed using the `docling` tool.
```

Comment:
I think you only have to assume the format, not necessarily that it was processed via docling.

Reply:
While it may be true, it's not a strict requirement.

Reply:
Yes, I think that's right, but the format is the output format of docling, not any JSON file.  It needs to match the docling JSON schema so we can use hierarchical chunking, at least for now.  We can add support for other formats in the future.

Reply:
I can clarify that it follows the `docling` format because we should use the `instructlab-sdg` package that depend on that tool. As soon as they switch to a different format/tool, we will be updated accordingly.